pilot/pkg/bootstrap/certs.go

```diff
@@ -312,3 +307,59 @@ func CheckCert(certPEM, caCert []byte) error {
 	}
 	return nil
 }
+
+// InitCerts will create the certificates to be used by Istiod GRPC server and webhooks, signed by K8S server.
+func InitCerts(server *Server) {
+	if server.kubeClient == nil {
+		log.Warn("K8S not found, cert signing disabled.")
+		return
+	}
+	// TODO: fallback to citadel (or custom CA) if K8S signing is broken
+
+	// discAddr configured in mesh config - this is what we'll inject into pods.
+	discAddr := server.mesh.DefaultConfig.DiscoveryAddress
+	if IstiodAddress.Get() != "" {
+		discAddr = IstiodAddress.Get()
+	}
+	host, _, err := net.SplitHostPort(discAddr)
+	if err != nil {
+		log.Fatala("Invalid discovery address", discAddr, err)
+	}
+
+	hostParts := strings.Split(host, ".")
+
+	ns := "." + podNamespaceVar.Get()
+
+	// Names in the Istiod cert - support the old service names as well.
+	// The first is the recommended one, also used by Apiserver for webhooks.
+	names := []string{
+		hostParts[0] + ns + ".svc",
+		hostParts[0] + ns,
+		"istiod" + ns + ".svc",
+		"istio-pilot" + ns,
```

Comment:
Changed it a bit. 
It's tricky - K8S calls the inject webhook using the .svc - so this name is absolutely required in the cert. 

The other names are useful for migrating/backward compat. Will be removed once we get rid of the components and possibly wait 1 release.

Reply:
I think we should make the DiscoveryAddress follow the K8S convention - i.e. 
  istiod.istio-system.svc:15012

and get rid of the names without svc. This will mean a single name to rule them all.